datafusion/core/src/datasource/listing/table.rs

```diff
@@ -536,6 +537,23 @@ impl TableProvider for ListingTable {
             return Ok(Arc::new(EmptyExec::new(false, projected_schema)));
         }
 
+        // extract types of partition columns
+        let table_partition_cols = self
+            .options
+            .table_partition_cols
+            .iter()
+            .map(|col| {
+                (
+                    col.0.to_owned(),
+                    self.table_schema
+                        .field_with_name(&col.0)
+                        .unwrap()
```

Comment:
I think this is going to `panic` if the user specifies a partition column that is not present.

It would be nice to make it an error -- can you either do so as a follow on PR or file a ticket?